plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestTransactionLogAccess.java

```diff
@@ -956,6 +957,74 @@ public void testTableSnapshotsActiveDataFilesCacheDisabled()
                         .build());
     }
 
+    @Test
+    public void testLoadSnapshotWithEndVersion()
+            throws Exception
+    {
+        setupTransactionLogAccessFromResources("person", "databricks73/person");
+
+        String tableName = "person";
+        String tableDir = getClass().getClassLoader().getResource("databricks73/" + tableName).toURI().toString();
+        DeltaLakeConfig shortLivedActiveDataFilesCacheConfig = new DeltaLakeConfig();
+        shortLivedActiveDataFilesCacheConfig.setDataFileCacheTtl(new Duration(10, TimeUnit.MINUTES));
+        MetadataEntry metadataEntry = transactionLogAccess.getMetadataEntry(SESSION, tableSnapshot);
+        ProtocolEntry protocolEntry = transactionLogAccess.getProtocolEntry(SESSION, tableSnapshot);
+
+        // Version 10 has a checkpoint file
+        transactionLogAccess.flushCache();
+        assertFileSystemAccesses(
+                () -> {
+                    transactionLogAccess.loadSnapshot(SESSION, new SchemaTableName("schema", tableName), tableDir, Optional.of(9L));
+                    try (Stream<AddFileEntry> addFileEntries = transactionLogAccess.getActiveFiles(SESSION, tableSnapshot, metadataEntry, protocolEntry, TupleDomain.all(), alwaysTrue())) {
+                        assertThat(addFileEntries.count()).isEqualTo(12L);
+                    }
+                },
+                ImmutableMultiset.<FileOperation>builder()
+                        .add(new FileOperation("_last_checkpoint", "InputFile.newStream"))
+                        .add(new FileOperation("00000000000000000000.json", "InputFile.newStream"))
+                        .add(new FileOperation("00000000000000000001.json", "InputFile.newStream"))
+                        .add(new FileOperation("00000000000000000002.json", "InputFile.newStream"))
+                        .add(new FileOperation("00000000000000000003.json", "InputFile.newStream"))
+                        .add(new FileOperation("00000000000000000004.json", "InputFile.newStream"))
+                        .add(new FileOperation("00000000000000000005.json", "InputFile.newStream"))
+                        .add(new FileOperation("00000000000000000006.json", "InputFile.newStream"))
+                        .add(new FileOperation("00000000000000000007.json", "InputFile.newStream"))
+                        .add(new FileOperation("00000000000000000008.json", "InputFile.newStream"))
+                        .add(new FileOperation("00000000000000000009.json", "InputFile.newStream"))
+                        .add(new FileOperation("00000000000000000010.checkpoint.parquet", "InputFile.length"))
+                        .add(new FileOperation("00000000000000000010.checkpoint.parquet", "InputFile.newInput"))
```

Comment:
Why do we read the checkpoint for version 10 when loading the snapshot for version 9?